File: test-automation-core/src/main/java/com/nortal/test/core/rest/interceptors/LoggingInterceptor.java
package com.nortal.test.core.rest.interceptors;

import java.io.IOException;
import org.springframework.stereotype.Component;
import lombok.extern.slf4j.Slf4j;
import okhttp3.Interceptor;
import okhttp3.Response;

@Slf4j
@Component
public class LoggingInterceptor implements Interceptor {

	@Override
	public Response intercept(final Interceptor.Chain chain) throws IOException {
		var request = chain.request();

		var startTime = System.currentTimeMillis();

		var response = chain.proceed(request);

		var duration = System.currentTimeMillis() - startTime;
		log.info("Received response with code {} for {} in {}ms.", response.code(), response.request().url(), duration);

		return response;
	}

}

File: test-automation-core-old/src/main/java/com/nortal/test/core/rest/interceptors/LoggingInterceptor.kt
package com.nortal.test.core.rest.interceptors

import lombok.extern.slf4j.Slf4j
import okhttp3.Interceptor
import okhttp3.Response
import org.springframework.stereotype.Component
import java.io.IOException

@Slf4j
@Component
class LoggingInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val startTime = System.currentTimeMillis()
        val response = chain.proceed(request)
        val duration = System.currentTimeMillis() - startTime
        LoggingInterceptor.log.info("Received response with code {} for {} in {}ms.", response.code(), response.request().url(), duration)
        return response
    }
}